hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/ClientCacheStatisticsImpl.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.cache.impl;
+
+import com.hazelcast.cache.impl.CacheEntryCountResolver;
+import com.hazelcast.cache.impl.CacheStatisticsImpl;
+import com.hazelcast.monitor.NearCacheStats;
+
+/**
+ * {@link com.hazelcast.cache.CacheStatistics} implementation for client side {@link com.hazelcast.cache.ICache}.
+ */
+public class ClientCacheStatisticsImpl
+        extends CacheStatisticsImpl {
+
+    private NearCacheStats nearCacheStats;
+
+    public ClientCacheStatisticsImpl(long creationTime) {
+        super(creationTime, CacheEntryCountResolver.createEntryCountResolver());
+    }
+
+    public ClientCacheStatisticsImpl(long creationTime, NearCacheStats nearCacheStats) {
+        super(creationTime, CacheEntryCountResolver.createEntryCountResolver());
+        this.nearCacheStats = nearCacheStats;
+    }
+
+
+    @Override
+    public long getOwnedEntryCount() {
+        throw new UnsupportedOperationException("This statistic is not supported for client.");
+    }
+
+    @Override
+    public NearCacheStats getNearCacheStatistics() {
+        if (nearCacheStats != null) {
```

Comment:
I would flip the 2 statements. Normally you first check for the unhappy situation and then you continue to the real logic. For example argument validation.